pwndbg/commands/cymbol.py

```diff
@@ -0,0 +1,269 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+Add, load, show, edit, or delete symbols for custom structures.
+
+For the generation of the symbols g++/gcc is being used under the hood.
+
+In case of remote debugging a binary which is not native to your architecture it
+is advised to configure the 'gcc-config-path' config parameter to your own cross-platform
+gnu gcc compiled toolchain for your target architecture.
+
+You are advised to configure the 'cymbol-editor' config parameter to the path of your
+favorite text editor. Otherwise cymbol exapnds $EDITOR and $VISUAL environement varialbes
+to find the path to the default text editor.
+"""
+
+import argparse
+import os
+import subprocess
+import sys
+import tempfile
+
+import gdb
+
+import pwndbg
+import pwndbg.commands
+import pwndbg.gdblib.arch
+import pwndbg.lib.gcc
+import pwndbg.lib.tempfile
+from pwndbg.color import message
+
+gcc_compiler_path = pwndbg.gdblib.config.add_param(
+    "gcc-compiler-path",
+    "",
+    "Path to your own gnu gcc/g++ toolchain for generating imported symbols.",
+)
+
+cymbol_editor = pwndbg.gdblib.config.add_param(
+    "cymbol-editor", "", "Path to your editor of your choice for editing custom structures."
+)
+
+
+def OnlyWhenStructureExists(func):
+    def wrapper(custom_structure_name):
+        pwndbg_cachedir = pwndbg.lib.tempfile.cachedir("custom-symbols")
+        pwndbg_custom_structure_path = os.path.join(pwndbg_cachedir, custom_structure_name) + ".c"
+
+        if not os.path.exists(pwndbg_custom_structure_path):
+            print(message.error("No custom structure was found with the given name!"))
+            return
+
+        return func(custom_structure_name, pwndbg_custom_structure_path)
+
+    return wrapper
+
+
+def PromptForOverwrite(func):
```

Comment:
Does it make sense that this is a decorator? Wouldn't it be better to just execute this code in the currently wrapped command function?

Reply:
Hmm I thought this might be useful for the future in case of adding new flags to the command, but yeah probably I shouldn't implement it as a decorator in this case.